### PR TITLE
feat: add seed offset args to sampler to allow cuda graph support

### PR DESCRIPTION
**Pull request by ksukrit**
## 📌 Description

This PR adds optional seed/offset args to all the sampler functions to prevent calling the `get_seed_and_offset` function. If that function is not called, we can potentially make the sampler forward call as part of CUDAGraph and use that to replay it.

We can directly compute the Seed/offset values, before launching the graph in a similar way to as it is being done in the current method and pass them when making the flashinfer call

## 🔍 Related Issues
#978 : top_k_top_p_sampling_from_logits incompatible with torch.compile + CUDAGraph

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [x] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **New Features**
  * Optional seed and offset parameters added to sampling APIs to enable deterministic RNG control while remaining optional.

* **Tests**
  * New tests verify reproducible sampling when using the same seed/offset and variability when different values are used.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by yzh119 — COMMENTED** [review]
Hi @ksukrit it's a good feature to have.

Would you mind adding unittest for this?

Also, it would be to support an array of seed/offset, where each of them is a int64 array, to describe per-batch seed/offset. (It requires some modification to the kernels).

**Comment by ksukrit** [comment]
> Hi @ksukrit it's a good feature to have.
> 
> Would you mind adding unittest for this?
> 
> Also, it would be to support an array of seed/offset, where each of them is a int64 array, to describe per-batch seed/offset. (It requires some modification to the kernels).

Sure thing, will add the unittests for this. But I had a quick question, right now the kernels just take a single int offset/seed value right ? The main purpose of this was to avoid the `get_seed_and_offset` and offset call and allow cuda graph capture. 

Is it okay if I take up the seed/offset array changes for the batch in a separate PR @yzh119

**Comment by yzh119** [comment]
> Is it okay if I take up the seed/offset array changes for the batch in a separate PR

Sure, we can do that in a separate PR.
